packages/block-library/src/block/edit.js

```diff
@@ -84,6 +97,56 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 
 	const blockProps = useBlockProps();
 
+	// local states for gudie modal
+	const [ isGudieOpen, setIsGudieOpen ] = useState( false );
+
+	// Get all the unsaved Records
+	const dirtyRecords = useSelect( ( select ) => {
+		return select( coreStore ).__experimentalGetDirtyEntityRecords();
+	}, [] );
+
+	const selectedBlock = useSelect( ( select ) => {
+		return select( blockEditorStore ).getSelectedBlock();
```

Comment:
This is used to get `ref` later in code, but we already have a `ref` for the current block.